sphinx/writers/latex.py

```diff
@@ -1754,16 +1757,8 @@ def depart_image(self, node):
 
     def visit_figure(self, node):
         # type: (nodes.Node) -> None
-        ids = ''  # type: unicode
-        for id in self.pop_hyperlink_ids('figure'):
-            ids += self.hypertarget(id, anchor=False)
-        if node['ids']:
-            ids += self.hypertarget(node['ids'][0], anchor=False)
+        targets = self.hypertarget_to(node)
         self.restrict_footnote(node)
-        if (len(node.children) and
-           isinstance(node.children[0], nodes.image) and
-           node.children[0]['ids']):
-            ids += self.hypertarget(node.children[0]['ids'][0], anchor=False)
```

Comment:
Originally, this if-block was introduced to adjust `:name:` option of the `figure` directive. But it is not needed at present because we added patched `figure` directive to hack it (see `sphinx/directives/patches.py`).

Therefore, I removed this block simply.